src/pyhf/paramview.py

```diff
@@ -0,0 +1,69 @@
+from . import get_backend, default_backend, events
```

Comment:
`pnames` is not used for anything but inside `ParamViewer`. Given that this is the case, we should move the logic of building `pnames` into `ParamViewer`... in fact, thinking about this, ParamViewer relies on the `pdfconfig` and the `batch_size` only. One could imagine actually moving some of this out of the constraints entirely and just building a `ParamViewerFactory` that only takes in the `pdfconfig` and then you can configure it with the batch size after the fact, since it should be pretty deterministic.